### PR TITLE
Refactors Terraform plan to break out steps

### PR DESCRIPTION
**Pull request by dmccown**
This change should make it easier to work with a running environment.
You may now apply changes without doing a full teardown/rebuild any time
you make a change to the terraform plan.

Adds some documenation on how to use the Terraform tooling.

Signed-off-by: David McCown <dmccown@chef.io>

### CODE REVIEW

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Testing Against Azure - Integration Testing
+
+## Problem Statement
+
+We want to test Azure-related InSpec resource against Azure itself. This requires a test fixture in Azure to exmine using InSpec.
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: s/exmine/examine/

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## General Approach
+
+We have a Terraform plan to setup and destroy test fixtures in Azure. When the environment is running we have a set of integration tests that may run against it.
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: s/setup/set up/

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Credentials are handled via a `~/.azure/credentials` file.
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: This is a bit mysterious. Can we offer readers a clue about where/how to get the info that goes in there? What it looks like?

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+### Installing Terraform
+
+Download [Terraform](https://www.terraform.io/downloads.html).  We require at least v0.10 . To install and choose from multiple Terraform versions, consider using [tfenv](https://github.com/kamatama41/tfenv).
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: s/10 \. /10. /

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Current Solution
+
+In Azure we have setup an application. We use the credentials from this application to connect to Azure. From there we use Terraform to create the needed resources that we test against.
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: "In Azure we have setup an application." →"We have set up an application in Azure."

File: test/integration/azure/TESTING_AGAINST_AZURE.md (hunk 3 above)
Comment (dmccown) [reply test/integration/azure/TESTING_AGAINST_AZURE.md]: Yeah I'll do that and give more information on setting up the account you'll need.

File: test/integration/azure/TESTING_AGAINST_AZURE.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+### Creating the Application account
+
+
+
+## Running the integration tests
+`INSPEC_TERRAFORM_ENVIRONMENT` should be set to a unique for you to work against. See [Terraform Workspaces](https://www.terraform.io/docs/state/workspaces.html)
```
Comment (TrevorBramble) [inline test/integration/azure/TESTING_AGAINST_AZURE.md]: s/a unique for/a unique value for/

File: test/integration/azure/TESTING_AGAINST_AZURE.md (hunk 3 above)
Comment (dmccown) [reply test/integration/azure/TESTING_AGAINST_AZURE.md]: Yeah I'll do that and give more information on setting up the account you'll need.

File: Rakefile [hunk 7]
```diff
@@ -157,38 +157,71 @@ namespace :test do
...
-      # Dump TF outputs to InSpec attributes file
+      Rake::Task["test:azure:vars"].execute
+      Rake::Task["test:azure:plan"].execute
+      Rake::Task["test:azure:apply"].execute
+    end
+
+    desc "Generate terraform.tfvars file"
+    task :vars do |t, args|
+
+      unless File.exist?("#{integration_dir}/build/terraform.tfvars")
```
Comment (TrevorBramble) [inline Rakefile]: Maybe treat as guard statement and use `next if` instead, so the entire contents of the task doesn't have to indent under the conditional?

File: Rakefile [hunk 8]
```diff
@@ -157,38 +157,71 @@ namespace :test do
...
+          subscription_id = "#{creds[:subscription_id]}"
+          client_id = "#{creds[:client_id]}"
+          client_secret = "#{creds[:client_secret]}"
+          tenant_id = "#{creds[:tenant_id]}"
+          storage_account_name = "#{sa_name}"
+          admin_password = "#{admin_password}"
+          suffix = "#{suffix}"
+        VARS
+
+        File.write("#{integration_dir}/build/terraform.tfvars", content)
```
Comment (TrevorBramble) [inline Rakefile]: This string is inlined in two places, maybe define in one place at the start?

**Review by TrevorBramble — APPROVED** [review]
Looks great!

**Review by pwelch — APPROVED** [review]

**Review by miah — APPROVED** [review]
Solid improvement. Thanks!

**Review by jquick — APPROVED** [review]
Thanks @dmccown !

**Pull request merged**